plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/action/NewPackageAction.java

```diff
@@ -41,13 +42,37 @@
  */
 @Singleton
 public class NewPackageAction extends AbstractNewResourceAction {
+
+    private final AppContext appContext;
     private final InputValidator nameValidator = new NameValidator();
 
     @Inject
-    public NewPackageAction(JavaResources javaResources, JavaLocalizationConstant localizationConstant) {
+    public NewPackageAction(JavaResources javaResources,
+                            JavaLocalizationConstant localizationConstant,
+                            AppContext appContex    ) {
         super(localizationConstant.actionNewPackageTitle(),
               localizationConstant.actionNewPackageDescription(),
               javaResources.packageIcon());
+        this.appContext = appContex;
+    }
+
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent e) {
+        CurrentProject project = appContext.getCurrentProject();
+        if (project == null || !"maven".equals(project.getRootProject().getType())) {
```

Comment:
We have special constant for Maven PType ID:
https://github.com/codenvy/che-plugins/blob/master/plugin-java/che-plugin-java-ext-maven/src/main/java/org/eclipse/che/ide/extension/maven/shared/MavenAttributes.java#L17